### PR TITLE
[ADF-5026] upgrade to Angular 9

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- migrate to Angular Material imports
- upgrade to Angular 9
- upgrade Material Components
- fix incorrect `@angular/router` imports in demo shell
- fix incorrect exports
- fix ADF CLI build and scripts
- cleanup providers (remove unused declarations for those provided in root)
- remove unused dependencies
- support `ng build testing` for the testing library

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-5026

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -85,72 +84,63 @@
...
     "@ngx-translate/core": "^11.0.0",
-    "@types/request": "^2.48.3",
     "adf-monaco-extension": "0.0.8",
     "adf-tslint-rules": "0.0.6",
     "chart.js": "2.5.0",
-    "classlist.js": "1.1.20150312",
-    "core-js": "^2.5.4",
-    "custom-event-polyfill": "0.3.0",
-    "hammerjs": "2.0.8",
+    "hammerjs": "^2.0.8",
```
Comment (popovicsandras) [inline package.json]: Do we need this hammerjs at the end? I saw it was removed from everywhere except this one.

Comment (DenysVuika) [reply package.json]: We need it only on the application level. We have no api or components that are actually depending on the hammer in ADF

Comment (popovicsandras) [reply package.json]: 👍

**Comment by petebacondarwin** [comment]
This PR is essential for the library to work with v9.0.0 of Angular Material due to the breaking change:

> Components can no longer be imported through “@angular/material”. Use the individual secondary entry-points, such as @angular/material/button.

See failing validation tests here: https://app.circleci.com/jobs/github/angular/ngcc-validation/3399

**Comment by eromano** [comment]
thanks @petebacondarwin we are working on it because there are other app depending on this library we are to take in count probably to release a new version of ADF with a major version change.
That update should be due for ADF version 4 Usually we also remove the deprecated code in the major release so this will take a little longer time in order to complete.

**Pull request closed without merging**